support/cas-server-support-ldap/src/main/java/org/apereo/cas/config/LdapAuthenticationConfiguration.java

```diff
@@ -154,9 +156,22 @@ public PrincipalFactory ldapPrincipalFactory() {
         return PrincipalFactoryUtils.newPrincipalFactory();
     }
 
+    @Bean
+    SetFactoryBean ldapAuthenticationHandlerFactoryBean() {
+        return new SetFactoryBean() {
+            @Override
+            protected void destroyInstance(final Set set) {
+                set.forEach(handler -> {
+                    ((LdapAuthenticationHandler) handler).destroy();
```

Comment:
Can the cast here be more generic to use DisposableBean? Casting to a concrete impl *might* cause issues for extensions.

Reply:
No problem